pygbe/main.py

```diff
@@ -355,14 +367,27 @@ def main(argv=sys.argv, log_output=True, return_output_fname=False,
     print('-'*30)
     print('Total setup time   : {}s'.format(setup_time))
 
-    tic = time.time()
+
+    #   Check if there is a complex dielectric
+    complexDiel = 0
+    for f in field_array:
+        if numpy.iscomplexobj(f.E):
+            complexDiel = 1
+
 
     ### Solve
+    tic = time.time()
+
     print('Solve')
-    phi = numpy.zeros(param.Neq)
+    # Initializing phi dtype according to the problem we are solving.
+    if complexDiel == 1:
+        phi = numpy.zeros(param.Neq, type(f.E))
```

Comment:
Inside a list comprehension, `f` is a proper temporary variable and doesn't exist outside of the comprehension, however variables defined as iterators over for loops stick around.

```python
In [1]: [i for i in range(3)]
Out[1]: [0, 1, 2]

In [2]: print(i)
-------------------------------------------------------------------------
NameError                               Traceback (most recent call last)
<ipython-input-2-ed403fc09801> in <module>()
----> 1 print(i)

NameError: name 'i' is not defined

In [3]: for i in range(3):
   ...:     print(i)
   ...:     
0
1
2

In [4]: i
Out[4]: 2
```